core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -113,5 +214,46 @@ class LogCleanerManagerTest extends JUnitSuite {
     log
   }
 
+  /*
+   * Compute the first segment in the tail that contains the specified size
+   */
+  def computeSizeLagSegment(log: Log, byteLag: Long): LogSegment = {
+    var lagSum = 0L
+    var lagSegment: LogSegment = null
+    log.logSegments.toSeq.reverse.toStream.takeWhile(_ => lagSum < byteLag).foreach {
+      s =>
+        lagSum += s.size
+        lagSegment = s
+    }
+    lagSegment
+  }
+
+  def makeLog(dir: File = logDir, config: LogConfig = logConfig) =
```

Comment:
Instead of duplicating makeLog, readFromLog and writeToLog (which are already in ClearnerTest) here, could we put them in a LogTestUtils class and share?